subprojects/signer/src/main/kotlin/com/avito/plugin/SignServicePlugin.kt

```diff
@@ -21,71 +21,89 @@ public class SignServicePlugin : Plugin<Project> {
 
+            // AGP 4.2 remove this block completely
+            appExtension.applicationVariants.all { variant: com.android.build.gradle.api.ApplicationVariant ->
+
+                val buildTypeName = variant.buildType.name
+                val apkToken: String? = extension.apkSignTokens[buildTypeName]
+                val bundleToken: String? = extension.bundleSignTokens[buildTypeName]
+
+                variant.outputsAreSigned = apkToken.hasContent() || bundleToken.hasContent()
+
+                target.tasks.signedApkTaskProvider(variant.name).configure { signApkTask ->
+                    signApkTask.dependsOn(variant.packageApplicationProvider)
+                }
+
+                target.tasks.signedBundleTaskProvider(variant.name).configure { signBundleTask ->
+                    signBundleTask.dependsOn(target.tasks.bundleTaskProvider(variant))
+                }
+            }
+
+            // AGP 4.2 variant
+            // onVariants { variant: ApplicationVariant ->
             target.androidCommonExtension.onVariants {
+
                 val variant = this
 
                 val urlResolver = UrlResolver(extension)
 
                 registerTask<SignApkTask>(
                     tasks = target.tasks,
-                    variant = this,
-                    taskName = signApkTaskName(variant),
+                    variant = variant,
+                    taskName = signApkTaskName(variant.name),
                     extension = extension,
                     signingResolver = SigningResolver(
                         extension = extension,
-                        variant = this,
+                        variant = variant,
                         signTokensMap = extension.apkSignTokens
                     ),
                     urlResolver = urlResolver
                 )
 
                 registerTask<SignBundleTask>(
                     tasks = target.tasks,
-                    variant = this,
-                    taskName = signBundleTaskName(variant),
+                    variant = variant,
+                    taskName = signBundleTaskName(variant.name),
                     extension = extension,
                     signingResolver = SigningResolver(
                         extension = extension,
-                        variant = this,
+                        variant = variant,
                         signTokensMap = extension.bundleSignTokens
                     ),
                     urlResolver = urlResolver
                 )
             }
 
             target.androidCommonExtension.onVariantProperties {
+                val variant = this
 
-                artifacts.use(target.tasks.signedApkTaskProvider(this))
-                    .wiredWithDirectories(
-                        taskInput = SignApkTask::unsignedDirProperty,
-                        taskOutput = SignApkTask::signedDirProperty
-                    )
-                    .toTransform(ArtifactType.APK)
-
```

Comment:
here, you should be able to do :
artifacts.use(target.tasks.signedApkTaskProvider(this)).configure {
                    signedDirProperty.set(File("/path/to/where/you/want/your/signed/files"))
                }

same for bundle

Reply:
`artifacts.use` returns `TaskBasedOperation` which don't have `configure`; but probably you meant that i should configure our custom `signedApkTaskProvider`?

We overwrite final signed file in standard (AGP provided) location, i don't really know why so and can change this logic to output in custom directory and then wire signed file consumers accordingly.

@eugene-krivobokov maybe you remember why we need to override?

Reply:
No, I don't remember. 
I assume only for convenience, to have singed artifact in one place, regardless of this plugin.
When no other plugins are consuming this artifact it doesn't look important.

Reply:
yes my example was based on using the TaskProvider for your signing task.